src/sql/workbench/parts/objectExplorer/browser/connectionTreeAction.ts

```diff
@@ -41,40 +41,37 @@ export class RefreshAction extends Action {
 		super(id, label);
 		this._tree = tree;
 	}
-	public run(): Promise<boolean> {
+	public async run(): Promise<boolean> {
 		let treeNode: TreeNode;
 		if (this.element instanceof ConnectionProfile) {
 			let connection: ConnectionProfile = this.element;
 			if (this._connectionManagementService.isConnected(undefined, connection)) {
 				treeNode = this._objectExplorerService.getObjectExplorerNode(connection);
 				if (treeNode === undefined) {
-					this._objectExplorerService.updateObjectExplorerNodes(connection.toIConnectionProfile()).then(() => {
-						treeNode = this._objectExplorerService.getObjectExplorerNode(connection);
-					});
+					await this._objectExplorerService.updateObjectExplorerNodes(connection.toIConnectionProfile());
```

Comment:
Didn't we have issues with doing this before?

Reply:
Yeah this is a behavior change - I don't think we had any issues it was just that PR was a stale PR and I noticed i made a few mistakes before. Just redoing it.